lib/run/summary.js

```diff
@@ -13,19 +13,20 @@ var _ = require('lodash'),
 RunSummary = function RunSummary (emitter, options) {
     // keep a copy of this instance since, we need to refer to this from various events
     var summary = this,
-        onlyCLIReporterEnabled,
-        skipExecutionsSummary;
-
-    // Need to do add this check for unit-tests when no options are passed
-    if (options) {
-        if (!options.skipExecutionsSummary) {
-            onlyCLIReporterEnabled = (Array.isArray(options.reporters) &&
-                options.reporters.length === 1 &&
-                options.reporters.includes('cli'));
-        }
-        // skipExecutionsSummary in case the option is specified, or only cli is specified as reporter
-        skipExecutionsSummary = options.skipExecutionsSummary || onlyCLIReporterEnabled;
+        executionSummaryOptions = _.get(options, 'executionsSummary', true),
+        skipExecutionSummary;
+
+    if (!executionSummaryOptions) {
+        // set skipExecutionSummary to true in case `--no-execution-summary` option is specified
+        skipExecutionSummary = true;
```

Comment:
I think we can still get rid of this `if` block.

Reply:
Combining the `cli` only present as reporter check with the `!executionSummaryOptions` we can eliminate the block but I guess it'll make the code less readable. Please let me know if you've any other suggestions for this.

Reply:
How about this: https://github.com/postmanlabs/newman/pull/2509#discussion_r493198231

Single `if` block without combining.